include/pagmo/type_traits.hpp

```diff
@@ -193,6 +193,22 @@ using decay_t = typename std::decay<T>::type;
 template <bool B, typename T = void>
 using enable_if_t = typename std::enable_if<B, T>::type;
 
+/// SFINAE enabler for arithmetic types
```

Comment:
I'd move these enablers in the detail namespace, as they are not used in any public interface that I see and they are rather simple anyway.